File: 367.py
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        start, end = 0, num
        while start + 1 < end:
            mid = (end - start)/2 + start
            if mid ** 2 == num:
                return True
            if mid**2 > num:
                end = mid
            else:
                start = mid
        if start ** 2 == num or end ** 2 == num:
            return True
        else:
            return False

File: 367.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if (num < 2){
            return true;
        }
        int sub = 1;
        while(num > 0){
            num -= sub;
            sub += 2;
        }
        return num == 0;
    }
};
